### PR TITLE
Features rfc7231 methods

### PR DESCRIPTION
**Pull request by nyamsprod**
I've added the following methods to make the class even more useful:
- `Http::isInformational` tell whether the submitted code is an informational code
- `Http::isSuccessful` tell whether the submitted code is a successful code 
- `Http::isRedirection` tell whether the submitted code is a redirection code
- `Http::isClientError` tell whether the submitted code is a client error code
- `Http::isServerError` tell whether the submitted code is a server error code 
- `Http::isUnassigned` tell whether the submitted code is a unassigned code by IANA
- `Http::isCustom` tell whether the submitted code is considered custom in regard to RFC7231

Of course when used the developer is encourage to mix the response to have a better understanding of the status code. Hope you like the addition.

I have updated the Unit test accordingly

### CODE REVIEW

**Comment by lukasoppermann** [comment]
Hey @nyamsprod thanks for this nice contribution. I will review this later today.

File: src/Httpstatus.php [hunk 1]
```diff
@@ -157,15 +184,152 @@ public function __construct($statusArray = [])
...
+    /**
+     * Tell whether the submitted code is a custom status code
+     *
+     * @param int $code
+     *
+     * @return bool
+     */
+    public function isCustom($code)
+    {
+        return 600 <= $this->filterHttpStatusCode($code);
```
Comment (colinodell) [inline src/Httpstatus.php]: Status codes above 5xx aren't valid according to the RFC (see #8)

File: src/Httpstatus.php [hunk 2]
```diff
@@ -157,15 +184,152 @@ public function __construct($statusArray = [])
...
+    /**
+     * Tell whether the submitted code is a unused status code By IANA
+     *
+     * @param int $code
+     *
+     * @return bool
+     */
+    public function isUnused($code)
+    {
+        return 306 === $this->filterHttpStatusCode($code);
```
Comment (colinodell) [inline src/Httpstatus.php]: I'm not sure I like having a single method checking for a specific status code.  We don't have methods like `isOK()` or `isTemporaryRedirect()`, so I'd vote against something like this.

**Comment by nyamsprod** [comment]
@colinodell I could restrict/remove the isCustom method ... the goal was to tell the end user if the status code is informational/sucessful/etc... I'm all about restricting the valid range of status code. 

This mean that we can reduce the valid range from 100 -> 999 (now) to 100 -> 599 (if this is what you suggested). That way the isCustom method would be automatically removed

The isUnused method on the other hand is based on IANA status code list

**Comment by lukasoppermann** [comment]
I am not sure of how much value the is unused method is, as I would get the
same result using hasStatusCode.

**Comment by nyamsprod** [comment]
Ok so let's strip the isCustom and isUnused methods then :+1:

**Pull request closed without merging**